### PR TITLE
Add discover filters

### PR DESCRIPTION
**Pull request by jordanwade**
Adds discover filters to the discover page. 

TO TEST: 
Go to /discover and click around make sure everything feels pretty good. Core functionality should mostly mirror current https://christfellowship.church/discover.

image
image

### CODE REVIEW

**Review by rkdavidson — COMMENTED** [review]
Hey @jordanwade overall this is feeling pretty solid. I'll do a deeper dive into the code after pairing with Caleb here, but wanted to drop some quick quirks and visual feedback/suggestions.

**Search state quirks**
- Feels weird to get a filter stuck on... could clicking the current filter un-select it?
- Is the text search supposed to be cleared when changing filters? I know there's some existing funky behavior around this, and the separate screens thing — so it may be as intended.



**Visual suggestions**
CleanShot 2021-03-15 at 14 25 10@2x

File: pages/discover/index.js [hunk 1]
```diff
@@ -66,90 +74,103 @@ const Discover = () => {
...
+        mb="base"
+      >
+        Search
+      </SearchField>
+
+      {showEmptyState && searchVisible && (
+        <Box my="xxl" pb="xxl" textAlign="center">
+          <Box as="h2">Looks like we couldn't find any results</Box>
+          <Box mb="base">
+            Consider reducing the number of filters or modifing your search
```
Comment (rkdavidson) [inline pages/discover/index.js:94]: Small typo
```suggestion
            Consider reducing the number of filters or modifying your search
```

File: components/SearchField/SearchField.js [hunk 2]
```diff
@@ -2,9 +2,16 @@ import React from 'react';
 
 import { Box, Button, Icon, systemPropTypes, TextInput, utils } from 'ui-kit';
 
+import Styled from './SearchField.styles';
```
Comment (drewbarontini) [inline components/SearchField/SearchField.js:5]: Can we get a newline after this?

**_Nitpick Ninja_** ™️

File: pages/discover/index.js [hunk 3]
```diff
@@ -66,90 +75,103 @@ const Discover = () => {
...
+                        ? filters[0]?.title
+                        : filter.title,
+                    contentId:
+                      filterValues.contentId === filter.id
+                        ? filters[0]?.id
+                        : filter.id,
+                  });
+                  setSearchVisible(false);
+                  reset();
+                }}
```
Comment (drewbarontini) [inline pages/discover/index.js:150]: It might be nice to pull this out into a function for easier readability?

**Review by drewbarontini — COMMENTED** [review]
CleanShot 2021-03-18 at 07 39 04@2x

CleanShot 2021-03-18 at 07 39 56@2x

"Keep Talking" should be the page title. It looks like this happens for any of the dedicated pages that are more than one word, so we should format those.

**Review by rkdavidson — APPROVED** [review]
Overall feels pretty solid!

There's some quirks I'm sure you're aware of/haven't had a chance to attend to yet, but feels shippable at least to get into CF QA hands.

**Issues I noticed**
- When you enable a filter, then click to go to one of it's results, and go back — the filters state is reset, so it's a top level search again
- There's some awkward double-loading moments, but they're brief and possibly inevitable due to how many network calls there are
- Truly shadowless cards feel strange, but I get the constraints of this view and that it may be necessary. Have you tried `s` or `xs` shadows instead of `no` shadow? (Low priority)

**Comment by jordanwade** [comment]
@rkdavidson

- I totally agree with the double loading issue. I want to come back to that and see if we can fix that issue.
- I could probably add the shadows back now that I'm not using the Carousel. I'll do a follow up fix for that one just incase I'm wrong.

**Review by rkdavidson — APPROVED** [review]
@jordanwade This feels ready to merge whenever you are! (Actually meant to approve before, my bad 🙈 )

**Pull request merged**